server/src/main/java/org/elasticsearch/inference/ModelConfigurations.java

```diff
@@ -173,10 +202,14 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
         builder.field(TaskType.NAME, taskType.toString());
         builder.field(SERVICE, service);
         builder.field(SERVICE_SETTINGS, serviceSettings);
-        builder.field(TASK_SETTINGS, taskSettings);
+        builder.field(OLD_TASK_SETTINGS, taskSettings);
         if (chunkingSettings != null) {
             builder.field(CHUNKING_SETTINGS, chunkingSettings);
         }
+        if (params.paramAsBoolean(INCLUDE_PARAMETERS, true)) { // default true so that REST requests get parameters
```

Comment:
Why do we need `INCLUDE_PARAMETERS`? Don't we know this from the endpoint version? Also shouldn't we either show parameters or task settings?

Reply:
The reason we need this field here is because we don't want to write the `parameters` to the index, since we are going to write them as `task_settings`, but you're right that I missed filtering on `endpointVersion`.

Reply:
the code already has a mechanism to determine if the config is being written to an index or not. 

```
if (params.paramAsBoolean(USE_ID_FOR_INDEX, false)) {
```

ModelRegistry sets `USE_ID_FOR_INDEX = true` when storing the config in the index. Let's rename `USE_ID_FOR_INDEX` to `FOR_INDEX` and use the same flag